derive/src/to_delegate.rs

```diff
@@ -390,11 +393,20 @@ fn generate_where_clause_serialization_predicates(item_trait: &syn::ItemTrait) -
 			let mut bounds: Punctuated<syn::TypeParamBound, Token![+]> =
 				parse_quote!(Send + Sync + 'static);
 			// add json serialization trait bounds
-			if visitor.serialize_type_params.contains(&ty.ident) {
-				bounds.push(parse_quote!(_serde::Serialize))
-			}
-			if visitor.deserialize_type_params.contains(&ty.ident) {
-				bounds.push(parse_quote!(_serde::de::DeserializeOwned))
+			if client {
+				if visitor.serialize_type_params.contains(&ty.ident) {
+					bounds.push(parse_quote!(_serde::de::DeserializeOwned))
```

Comment:
Don't we need both bounds?

Reply:
I don't think so, that would cause a compilation error of the generated code. If it is a parameter type in the server that needs to be serialized it's a result type in the client that needs to be deserialized and vice versa. The only reason it would need both bounds is if the same type is used as a parameter and a return type, which is handled correctly.

Reply:
Fair enough, although:

> If it is a parameter type in the server that needs to be serialized it's a result type in the client that needs to be deserialized and vice versa.

This implies that the type is Ser+De anyway, but I guess relaxed bounds just on the client interface are a good thing.